contracts/HubPool.sol

```diff
@@ -181,6 +180,7 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
 
     // This function has permission to call onlyFromCrossChainAdmin functions on the SpokePool, so its imperative
     // that this contract only allows the owner to call this method directly or indirectly.
+    // todo: think about removing this function and add all required L1->L2 admin calls directly to the x_chain adapter.
```

Comment:
I think the idea was to not allow the HubPool owner to relay any arbitrary `functionData` message via the Adapter, and implement all the specific cross-chain-admin functions directly on the Adapter, but I also think we decided this wasn't an issue since any `functionData` needs to be able to execute on the Adapter so there isn't the risk of arbitrary execution